Insufficient test isolation for default config file

**Describe the bug**
The `tmp_path_home` fixture does not isolate tests from local state correctly. An existing or broken `~/.config/instructlab/config.yaml` can affect behavior. This problem does not show up on CI and my local testing, because neither has a broken config file.

**To Reproduce**
1. `rm ~/.config/instructlab/config.yaml`
2. `touch ~/.config/instructlab/config.yaml`
3. `tox -e py311-unit`

**Expected behavior**
Tests don't use default config.

**Screenshots**
```
FAILED tests/test_lab.py::test_ilab_cli_help[('config',)] - AssertionError: 
FAILED tests/test_lab.py::test_ilab_cli_help[('config', 'init')] - AssertionError: 
FAILED tests/test_lab.py::test_ilab_cli_help[('model',)] - AssertionError: 
FAILED tests/test_lab.py::test_ilab_cli_help[('data',)] - AssertionError: 
FAILED tests/test_lab.py::test_ilab_cli_help[('system',)] - AssertionError: 
FAILED tests/test_lab.py::test_ilab_cli_help[('system', 'info')] - AssertionError: 
FAILED tests/test_lab.py::test_ilab_cli_help[('taxonomy',)] - AssertionError: 
FAILED tests/test_lab.py::test_ilab_cli_debug_params[('config', 'init')] - AssertionError: 
FAILED tests/test_lab.py::test_ilab_cli_debug_params[('system', 'info')] - AssertionError: 
FAILED tests/test_lab.py::test_ilab_missing_config[('config',)] - AssertionError: <Result TypeError('instructlab.configuration.Config() argument after ** must be a mapping, not NoneType')>
FAILED tests/test_lab.py::test_ilab_missing_config[('config', 'init')] - AssertionError: <Result TypeError('instructlab.configuration.Config() argument after ** must be a mapping, not NoneType')>
FAILED tests/test_lab.py::test_ilab_missing_config[('config', 'show')] - AssertionError: <Result TypeError('instructlab.configuration.Config() argument after ** must be a mapping, not NoneType')>
FAILED tests/test_lab.py::test_ilab_missing_config[('model',)] - AssertionError: <Result TypeError('instructlab.configuration.Config() argument after ** must be a mapping, not NoneType')>
FAILED tests/test_lab.py::test_ilab_missing_config[('model', 'chat')] - AssertionError: <Result TypeError('instructlab.configuration.Config() argument after ** must be a mapping, not NoneType')>
FAILED tests/test_lab.py::test_ilab_missing_config[('model', 'convert')] - AssertionError: <Result TypeError('instructlab.configuration.Config() argument after ** must be a mapping, not NoneType')>
FAILED tests/test_lab.py::test_ilab_missing_config[('model', 'download')] - AssertionError: <Result TypeError('instructlab.configuration.Config() argument after ** must be a mapping, not NoneType')>
FAILED tests/test_lab.py::test_ilab_missing_config[('model', 'evaluate')] - AssertionError: <Result TypeError('instructlab.configuration.Config() argument after ** must be a mapping, not NoneType')>
FAILED tests/test_lab.py::test_ilab_missing_config[('model', 'serve')] - AssertionError: <Result TypeError('instructlab.configuration.Config() argument after ** must be a mapping, not NoneType')>
FAILED tests/test_lab.py::test_ilab_missing_config[('model', 'test')] - AssertionError: <Result TypeError('instructlab.configuration.Config() argument after ** must be a mapping, not NoneType')>
FAILED tests/test_lab.py::test_ilab_missing_config[('model', 'train')] - AssertionError: <Result TypeError('instructlab.configuration.Config() argument after ** must be a mapping, not NoneType')>
FAILED tests/test_lab.py::test_ilab_missing_config[('model', 'list')] - AssertionError: <Result TypeError('instructlab.configuration.Config() argument after ** must be a mapping, not NoneType')>
FAILED tests/test_lab.py::test_ilab_missing_config[('data',)] - AssertionError: <Result TypeError('instructlab.configuration.Config() argument after ** must be a mapping, not NoneType')>
FAILED tests/test_lab.py::test_ilab_missing_config[('data', 'generate')] - AssertionError: <Result TypeError('instructlab.configuration.Config() argument after ** must be a mapping, not NoneType')>
FAILED tests/test_lab.py::test_ilab_missing_config[('system',)] - AssertionError: <Result TypeError('instructlab.configuration.Config() argument after ** must be a mapping, not NoneType')>
FAILED tests/test_lab.py::test_ilab_missing_config[('system', 'info')] - AssertionError: <Result TypeError('instructlab.configuration.Config() argument after ** must be a mapping, not NoneType')>
FAILED tests/test_lab.py::test_ilab_missing_config[('taxonomy',)] - AssertionError: <Result TypeError('instructlab.configuration.Config() argument after ** must be a mapping, not NoneType')>
FAILED tests/test_lab.py::test_ilab_missing_config[('taxonomy', 'diff')] - AssertionError: <Result TypeError('instructlab.configuration.Config() argument after ** must be a mapping, not NoneType')>
FAILED tests/test_lab_init.py::TestLabInit::test_init_noninteractive - AssertionError: assert 1 == 0
FAILED tests/test_lab_init.py::TestLabInit::test_init_interactive - AssertionError: assert 1 == 0
FAILED tests/test_lab_init.py::TestLabInit::test_init_interactive_git_error - AssertionError: assert 'Failed to clone taxonomy repo: Authentication failed' in 'You are using an aliased command, this will be deprecated in a future release. Please consider using `ilab config i...
FAILED tests/test_lab_init.py::TestLabInit::test_init_interactive_clone - AssertionError: assert 1 == 0
FAILED tests/test_lab_init.py::TestLabInit::test_init_interactive_with_preexisting_nonempty_taxonomy - AssertionError: assert 1 == 0
FAILED tests/test_lab_init.py::TestLabInit::test_init_interactive_with_preexisting_config - AssertionError: assert 1 == 0
========================================================================= 33 failed, 142 passed, 1 skipped, 2 warnings in 10.91s =========================================================================
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -123,6 +123,18 @@ jobs:
           python -m pip install --upgrade pip
           python -m pip install tox tox-gh>=1.2
 
+      # see https://github.com/instructlab/instructlab/issues/1886
+      - name: Verify test isolation
+        run: |
+          mkdir -p ~/.config/instructlab
+          echo "broken" > ~/.config/instructlab/config.yaml
+          
+          mkdir -p ~/.cache
+          echo "not a directory" > ~/.cache/instructlab
+          
+          mkdir -p ~/.local/share
+          echo "not a directory" > ~/.local/share/instructlab
+
       - name: Run unit and functional tests with tox
         run: |
           tox

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -60,16 +60,28 @@ def tmp_path_home(tmp_path: pathlib.Path) -> typing.Generator[pathlib.Path, None
     Yields `tmp_path` fixture path.
     """
     # First Party
+    from instructlab import lab
     from instructlab.configuration import DEFAULTS
 
+    # find the --config-file option
+    opt_config_file = [opt for opt in lab.ilab.params if opt.name == "config_file"][0]
+
     with mock.patch.dict(os.environ):
+        # set HOME as home of all XDG directories
         os.environ["HOME"] = str(tmp_path)
+        # unset custom config location
+        os.environ.pop(DEFAULTS.ILAB_GLOBAL_CONFIG, None)
+        # unset all XDG env vars
         for key in list(os.environ):
             if key.startswith("XDG_"):
                 os.environ.pop(key)
-
-        DEFAULTS._reset()  # resets the config defaults to use the new $HOME
-        yield tmp_path
+        # resets the config defaults to use the new $HOME
+        DEFAULTS._reset()
+        # patch default value of --config-file option
+        with mock.patch.object(
+            opt_config_file, attribute="default", new=DEFAULTS.CONFIG_FILE
+        ):
+            yield tmp_path
 
 
 @pytest.fixture

